pkg/cli/dump.go

```diff
@@ -198,8 +202,9 @@ func dumpTable(w io.Writer, conn *sqlConn, origDBName, origTableName string) err
 		inserts := make([][]string, 0, limit)
 		i := 0
 		for i < limit {
-			vals := make([]driver.Value, len(cols)+len(pkcols))
-			if err := rows.Next(vals); err == io.EOF {
+			rows.Next()
```

Comment:
this instance of `rows` (created at L195) doesn't get a defer. And we definitely have return statements in the middle.